thunder/core/interpreter.py

```diff
@@ -397,8 +399,11 @@ def __init__(
         self._callbacks: dict[INTERPRETER_CALLBACKS, Callable] = callbacks
         self._with_provenance_tracking = with_provenance_tracking
         if with_provenance_tracking:
-            assert isinstance(uncacheable_classes, (list, tuple))
-            uncacheable_classes = tuple(set(uncacheable_classes) | {NoneType, int, str, float, bool})
+            if uncacheable_classes is None:
+                uncacheable_classes = ()
+            uncacheable_classes = tuple(
+                set(uncacheable_classes) | {NoneType, int, str, float, bool, complex, torch.Tensor}
+            )
```

Comment:
On looking at this a bit further, I think your previous fix to add `complex` in `interpreter.py` was correct.

My understanding is that `interpreter.py` only understand basic Python and nothing about external libraries like PyTorch. `jit_ext.py` extends the capability to teach it about PyTorch support. So, I don't think `torch.Tensor` should be present here but instead be passed from `jit_ext.py`.

Sorry for the confusion.